asserts/device_asserts_test.go

```diff
@@ -166,6 +167,92 @@ func (mods *modelSuite) TestDecodeRequiredSnapsAreOptional(c *C) {
 	c.Check(model.RequiredSnaps(), HasLen, 0)
 }
 
+func (mods *modelSuite) TestDecodeValidatesSnapNames(c *C) {
+	withTimestamp := strings.Replace(modelExample, "TSLINE", mods.tsLine, 1)
+	encoded := strings.Replace(withTimestamp, reqSnaps, "required-snaps:\n  - foo_bar\n  - bar\n", 1)
+	a, err := asserts.Decode([]byte(encoded))
+	c.Assert(a, IsNil)
+	c.Assert(err, ErrorMatches, `assertion model: invalid snap name in header "required-snaps": foo_bar`)
```

Comment:
A native speaker may chime in here (@degville ?), but I'd probably say:
`assertion model: invalid snap name in "required-snaps" header: foo_bar` (note the header word after its name). Not totally sure though.

Reply:
+1 to putting header afterwards.